javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1194 @@
+      if (dest.hasNext) {
+        dest.next()
+        dest.set(source.next())
+      } else
+        throw new IndexOutOfBoundsException
+    }
+  }
+
+  private def rotateImpl[T](list: List[T], distance: Int): Unit = {
+    val listSize = list.size
+    if (listSize > 1 && distance % listSize != 0) {
+      def exchangeRotation(): Unit = {
+        def indexModulo(i: Int): Int =  modulo(i, listSize)
+        @tailrec
+        def rotateNext(cycleStartIndex: Int, count: Int, index: Int, value: T): Unit = {
+          val nextValue = list(index)
+          val newCount = count + 1
+          list(index) = value
+          if (index != cycleStartIndex) {
+            rotateNext(cycleStartIndex, newCount, indexModulo(index + distance), nextValue)
+          } else if (newCount < listSize) {
+            val nextCycleStart = cycleStartIndex + 1
+            rotateNext(nextCycleStart, newCount, indexModulo(nextCycleStart + distance),
+              list(nextCycleStart))
+          }
+        }
+        rotateNext(0, 0, indexModulo(distance), list(0))
+      }
+
+      def splitReverseRotation(): Unit = {
+        val splitPoint = modulo(-distance, listSize)
+        reverse(list.subList(0, splitPoint))
+        reverse(list.subList(splitPoint, listSize))
+        reverse(list)
+      }
+
+      list match {
+        case _: RandomAccess    => exchangeRotation()
+        case _ if listSize < 16 => exchangeRotation() // TODO benchmark and set proper limit
+        case _                  => splitReverseRotation()
+      }
+    }
+  }
+
+  private object WrappedCollections {
+
+    private[Collections] trait WrappedCollection[E, Coll <: Collection[E]] extends Collection[E] {
+      protected def inner: Coll
+
+      def size(): Int =
+        inner.size
+
+      def isEmpty: Boolean =
+        inner.isEmpty
+
+      def contains(o: Any): Boolean =
+        inner.contains(o)
+
+      def iterator(): Iterator[E] =
+        inner.iterator
+
+      def toArray: Array[AnyRef] =
+        inner.toArray()
+
+      def toArray[T <: AnyRef](a: Array[T]): Array[T] =
+        inner.toArray[T](a)
+
+      def add(e: E): Boolean =
+        inner.add(e)
+
+      def remove(o: Any): Boolean =
+        inner.remove(o)
+
+      def containsAll(c: Collection[_]): Boolean =
+        inner.containsAll(c)
+
+      def addAll(c: Collection[_ <: E]): Boolean =
+        inner.addAll(c)
+
+      def removeAll(c: Collection[_]): Boolean =
+        inner.removeAll(c)
+
+      def retainAll(c: Collection[_]): Boolean =
+        inner.retainAll(c)
+
+      def clear(): Unit =
+        inner.clear()
+
+      override def equals(obj: Any): Boolean =
+        inner.equals(obj)
+
+      override def hashCode(): Int =
+        inner.hashCode
+
+      override def toString: String =
+        inner.toString
+    }
+
+    private[Collections] trait WrappedSet[E, Coll <: Set[E]] extends WrappedCollection[E, Coll]
+                                                                with Set[E]
```

Comment:
We've deprecated the style with `with` aligned with `extends`. Wrap before `extends` and indent 4 spaces instead.